lib/components/resonator.ts

```diff
@@ -0,0 +1,25 @@
+import { frequency, capacitance } from "circuit-json"
+import {
+  type CommonComponentProps,
+  commonComponentProps,
+  lrPins,
+} from "lib/common/layout"
+import { expectTypesMatch } from "lib/typecheck"
+import { z } from "zod"
+
+export interface ResonatorProps extends CommonComponentProps {
+  frequency: number | string
+  loadCapacitance: number | string
+  pinVariant: "3pin"
+}
+
+export const resonatorProps = commonComponentProps.extend({
+  frequency: frequency,
+  loadCapacitance: capacitance,
+  pinVariant: z.literal("3pin"),
+})
+
+export const resonatorPins = lrPins
```

Comment:
This should be removed or changed to reflect the 3 pins, lrPins refers to left/right pins which isnt appropriate for a 3 pin circuit.
```suggestion
```